DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc

```diff
@@ -947,8 +947,12 @@ void Converter<PartSelector>::operator()(xml_h element) const {
            "+++ PartSelector for %s path: %s",
            specParName.c_str(),
            path.c_str());
-  if (path.substr(0, 3) == std::string(".*:")) {
-    path = path.substr(3);
+  auto npos = path.find(':');
+  if (npos != path.npos) {
+    std::string anyns{"//.*"};
+    if (path.substr(0, npos) == anyns) {
+      path = string("//").append(path.substr(npos + 1));
```

Comment:
I believe a few additional `std::string` creations could be avoided with something along
```suggestion
    constexpr std::string_view anyns{"//.*"sv};
    std::string_view pathView{path};
    if (pathView.substr(0, npos) == anyns) {
      path = string("//").append(pathView.substr(npos + 1));
```

Reply:
agree

Reply:
done